app/post.go

```diff
@@ -797,15 +797,22 @@ func (a *App) parseAndFetchChannelIdByNameFromInFilter(channelName, userId, team
 }
 
 func (a *App) searchPostsInTeam(teamId string, userId string, paramsList []*model.SearchParams, modifierFun func(*model.SearchParams)) (*model.PostList, *model.AppError) {
-	channels := []store.StoreChannel{}
+	channels := []chan store.StoreResult{}
 
 	for _, params := range paramsList {
 		// Don't allow users to search for everything.
 		if params.Terms == "*" {
 			continue
 		}
 		modifierFun(params)
-		channels = append(channels, a.Srv.Store.Post().Search(teamId, userId, params))
+
+		pchan := make(chan store.StoreResult, 1)
+		go func() {
+			postList, err := a.Srv.Store.Post().Search(teamId, userId, params)
+			pchan <- store.StoreResult{Data: postList, Err: err}
+			channels = append(channels, pchan)
```

Comment:
This append should be out of the go block, i'm not sure if this could be a problem but for sure is not necessary to do it there and I prefer to reduce the shared variables to the minimum.

Reply:
Indeed, this is a race condition that will result in `channels` either being wrong or altogether corrupted.

One option is to make `channels` just a `chan` itself (sized to `len(paramsList)`) and just send the result, but someone still needs to `close` the channel. I might suggest using a `WaitGroup`, and after waiting for all goroutines to exit, `close(channels)` in the main thread and then range through channels over the results.

Reply:
@jespino @lieut-data I just pushed the buffered channel with waitgroup fix, However build fails at i18n check. how do i fix that error?